### PR TITLE
Add links to GitLab

### LINKED ISSUE #867: link to gitlab from renkulab nav bar
## Motivation

It's hard for users to find the link to gitlab when they haven't created any projects yet, and this is a thing they have to do to be able to push existing projects to renkulab (because they have to set up an ssh key from gitlab).

## Proposal

put the link to gitlab in the nav bar (or somewhere else global)

### PR DESCRIPTION
**Pull request by ciyer**
Add links to the following GitLab URLs:

- User Profile
- User Settings
- GitLab Home

image

Can be tested at https://sekhar.dev.renku.ch/

Fix #867

### CODE REVIEW

**Comment by emmjab** [comment]
Thanks @ciyer !! I think this is good for now!

Slightly worried that the GitLab icon will be hard to describe for new-to-gitlab users, but we can include the icon in the docs I guess. It does do a good job saving space, and the first link once you click it does say GitLab.

Just noticed from testing that the help menu doesn't have a link to the docs~ but that's in another PR, will comment there.

**Comment by vfried** [comment]
I think that having the link to profile and settings under the user and under the gitlab sign could be a bit redundant, especially since both links are right. next to each other... 

image

image

Shall we remove some of this links somewhere?

File: src/utils/UIComponents.js [hunk 1]
```diff
@@ -568,24 +599,47 @@ function IconLink(props) {
   </span>;
 }
 
+function ExternalIconLinkWithTooltip(props) {
+  // eslint-disable-next-line no-unused-vars
```
Comment (vfried) [inline src/utils/UIComponents.js:603]: You can remove the  //eslint-disable.... if you remove also the setter since you are not using it.

**Comment by vfried** [comment]
> Thanks @ciyer !! I think this is good for now!
> 
> Slightly worried that the GitLab icon will be hard to describe for new-to-gitlab users, but we can include the icon in the docs I guess. It does do a good job saving space, and the first link once you click it does say GitLab.
> 
> Just noticed from testing that the help menu doesn't have a link to the docs~ but that's in another PR, will comment there.

I am all in for helping users have an easy beginning with renku, but i think we ned to draw a line about how much we "hide" gitlab inside renku, since for other users having gitlab is also seen as a benefit. I think that at some point someone using renku that has no idea about gitlab will have to make a quick search to see what it it.

File: src/utils/UIComponents.js (hunk 1 above)
Comment (ciyer) [reply src/utils/UIComponents.js]: Good suggestion! I have made this improvement.

**Comment by emmjab** [comment]
> I am all in for helping users have an easy beginning with renku, but i think we ned to draw a line about how much we "hide" gitlab inside renku, since for other users having gitlab is also seen as a benefit. I think that at some point someone using renku that has no idea about gitlab will have to make a quick search to see what it it.

wait i think we're on the same side? 😅 
basically what i mean is that it's hard for someone to google for an icon -- my suggestion would be to replace the fox icon with the word "GitLab"

> I think that having the link to profile and settings under the user and under the gitlab sign could be a bit redundant, especially since both links are right. next to each other...

Also confused --> before the "Settings" and "Profile" were **renku** settings and profile in the "User" menu.

**Comment by ciyer** [comment]
> I think that having the link to profile and settings under the user and under the gitlab sign could be a bit redundant, especially since both links are right. next to each other...
> 
> image
> image
> 
> Shall we remove some of this links somewhere?

You are right. In this case, since they are so close to each-other, it's confusing to have the links redundantly. I have removed them from the user menu and just left them in the GitLab menu.

**Comment by ciyer** [comment]
> Also confused --> before the "Settings" and "Profile" were **renku** settings and profile in the "User" menu.

There is no difference ATM between Renku Settings and GitLab settings, so those links lead to the same place.

**Comment by vfried** [comment]
Cool, i think it's better if we only have them in one place.

File: src/landing/NavBar.js [hunk 2]
```diff
@@ -49,6 +50,20 @@ class RenkuNavBar extends Component {
   }
 }
 
+function gitLabSettingsUrlFromProfileUrl(webUrl) {
+  // Yes, the settings URL ends with 'profile'; the profile URL ends with the username
```
Comment (lorenzo-cavazzi) [inline src/landing/NavBar.js:54]: :man_facepalming:

**Pull request merged**